### DISCUSSION TITLE
Artifactory with AWS ECR limits

### DISCUSSION THREAD

**Discussion by tonnico** [renovatebot/renovate#18971] [Request Help]
Hi,

in https://github.com/renovatebot/renovate/blob/fdf335043d0c9d90f757cad59e521ea1b57c9ae5/lib/modules/datasource/docker/index.ts#L832-L835 i found the limit of max 1000 tags per call. I hit the same limit when the ECR is behind an artifactory.

Is there some workaround so my `public-aws-ecr.example.com/image` will be detected like `public.ecr.aws/image`?

**Comment by tonnico** [comment]
Finally I found a small hack:
   ```js
 regexManagers: [
        {
            // initial digest pinning for regex manager https://github.com/renovatebot/renovate/issues/10993
            fileMatch: ['Dockerfile'],
            matchStrings: [
                'FROM public-aws-ecr\\.example\\.com/(?<depName>.*?)(?::(?<currentValue>.*?))?(?:@(?<currentDigest>sha256:[a-z0-9]+))?\\s',
            ],
            versioningTemplate: 'docker',
            datasourceTemplate: 'docker',
            depNameTemplate: 'public.ecr.aws/{{{depName}}}',
        },
```

**Reply by viceice** [reply]
probably better use `packageNameTemplate`. so you see the real name in PR but renovate uses packageName for lookup.

maybe registry aliases are another way to configure for you.

**Reply by tonnico** [reply]
Good point with the `packageNameTemplate`. 

Something like this in the `packageRules`-Section?
```
            matchManagers: ['dockerfile'],
            matchPackagePrefixes: ['public-aws-ecr.example.com'],
            registryAliases: ['public.ecr.aws'],
```

The documentation for registryAliases is relatively sparse. Unfortunately this did not work and still uses the example.com registry url.

**Reply by tonnico** [reply]
Somewhere deep in an old merge commit I found a small example. This should not be in the `packageRules`.

```
'dockerfile': {
            'registryAliases': {'public-aws-ecr.example.com': 'public.ecr.aws'}
}
```

which makes sense. Would be nice to have some kind of this example in the documentation.
Looks like that this will work. Currently I have still issues, but not at this point anymore.

**Reply by viceice** [reply]
feel free to extend the docs. 🤗

**Reply by tonnico** [reply]
@viceice sure. Can you please confirm, that the rule should not be used from the `packageRules` section?
